addons/website_slides/models/slide_channel.py

```diff
@@ -59,6 +78,42 @@ def _recompute_completion(self):
             completed_records._set_as_completed()
             completed_records._send_completed_mail()
 
+    @api.model
+    def create(self, values):
+        """
+        Override create method :
+        If an attendee is created as enrolled, subscribe its partner to the channel.
+        If an attendee is created for a partner without user, make it pending and send an invitation email.
+        """
+        channel_partners = super(ChannelUsersRelation, self).create(values)
+
+        # Mail Context For Invitation Mails
+        template = self.env.ref('website_slides.mail_template_slide_channel_invite', raise_if_not_found=False)
+        local_context = dict(
+            self.env.context,
+            default_use_template=bool(template),
+            default_template_id=template and template.id or False,
+            notif_layout='website_slides.mail_notification_channel_invite',
+        )
+        mail_values = []
+        invite_wizzard = self.env['slide.channel.invite'].with_context(local_context).create({'channel_id': channel_partners[0].channel_id.id})
+        subtype = self.env.ref('website_slides.mt_channel_slide_published', raise_if_not_found=False)
+
+        for channel_partner in channel_partners:
+            if channel_partner.enrolled:
+                # Subscribe enrolled partner if a user is linked to it
+                if channel_partner.partner_id.user_ids:
+                    channel_partner.channel_id.message_subscribe(partner_ids=channel_partner.partner_id.ids, subtype_ids=[subtype.id] if subtype else [])
+                # Make partner pending and send invitation mail (with password creation link)
+                else:
+                    channel_partner.enrolled = False
+                    invite_wizzard.write({'channel_id': channel_partner.channel_id.id})
+                    mail_values.append(invite_wizzard._prepare_mail_values(channel_partner))
+        invite_wizzard.sudo().unlink()
+        self.env['mail.mail'].sudo().create(mail_values)
+
+        return channel_partners
```

Comment:
I added an invite button to the view. It does look efficient. It uses the default channel filter, which corresponds to the case you come from a course A, click on attendees, then see only attendees of course A.
However, two changes come along the decision of removing this code:
1) When creating an attendee, it should be added to the followers of the channel. Therefore there is a short "create" override to do so
2) The idea was to allow inviting directly from the attendee tree view. It can be done through invite button BUT not inline anymore. This means that when creating an attendee inline, partners are filtered and only the ones having at least one user are selectable.